gadgets/bpfstats/test/unit/bpfstats_test.go

```diff
@@ -0,0 +1,151 @@
+}
+
+const (
+	testGadgetImage = "trace_open"
+)
+
+func TestBpfstatsGadget(t *testing.T) {
+	gadgettesting.InitUnitTest(t)
+	runnerConfig := &utilstest.RunnerConfig{}
+
+	testCases := map[string]testDef{
+		"by_gadget": {
+			runnerConfig:  runnerConfig,
+			generateEvent: generateEvent,
+			expectedEvents: []ExpectedBpfstatsEvent{
+				{
+					GadgetImage: gadgetrunner.GetGadgetImageName(testGadgetImage),
+					ProgID:      0,
+				},
+			},
+		},
+		"by_programs": {
+			runnerConfig:  runnerConfig,
+			generateEvent: generateEvent,
+			allPrograms:   true,
+			expectedEvents: []ExpectedBpfstatsEvent{
+				// programs from trace_open gadget. This introduces a dependency
+				// on the trace_open gadget, but it's almost guranteed that this
+				// gadget will have these two programs
+				{
+					GadgetImage: gadgetrunner.GetGadgetImageName(testGadgetImage),
+					ProgName:    "ig_openat_e",
+					ProgID:      utils.NormalizedInt,
+				},
+				{
+					GadgetImage: gadgetrunner.GetGadgetImageName(testGadgetImage),
+					ProgName:    "ig_openat_x",
+					ProgID:      utils.NormalizedInt,
+				},
+				// TODO: test external program
+			},
+		},
+	}
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			runner := utilstest.NewRunnerWithTest(t, testCase.runnerConfig)
+			globalParams := map[string]string{
+				"operator.ebpf.enable-bpfstats": "true",
+			}
+
+			normalizeEvent := func(event *ExpectedBpfstatsEvent) {
+				utils.NormalizeInt(&event.MapCount)
+				utils.NormalizeInt(&event.MapMemory)
+				utils.NormalizeInt(&event.Runcount)
+				utils.NormalizeInt(&event.Runtime)
+				utils.NormalizeInt(&event.ProgID)
+			}
+			onGadgetRun := func(gadgetCtx operators.GadgetContext) error {
+				utilstest.RunWithRunner(t, runner, func() error {
+					testCase.generateEvent(t)
+					return nil
+				})
+				return nil
+			}
+
+			var paramValues map[string]string
+			if testCase.allPrograms {
+				paramValues = map[string]string{"operator.ebpf.all": "true"}
+			}
+
+			opts := gadgetrunner.GadgetRunnerOpts[ExpectedBpfstatsEvent]{
+				Image:              "bpfstats",
+				Timeout:            5 * time.Second,
+				ParamValues:        paramValues,
+				GlobalParamsValues: globalParams,
+				OnGadgetRun:        onGadgetRun,
+				NormalizeEvent:     normalizeEvent,
+			}
+
+			gadgetRunner := gadgetrunner.NewGadgetRunner(t, opts)
+			gadgetRunner.RunGadget()
+
+			for _, expectedEvent := range testCase.expectedEvents {
+				expectedEvent.MapCount = utils.NormalizedInt
+				expectedEvent.MapMemory = utils.NormalizedInt
+				expectedEvent.Runcount = utils.NormalizedInt
+				expectedEvent.Runtime = utils.NormalizedInt
+
+				require.Contains(t, gadgetRunner.CapturedEvents, expectedEvent)
+			}
+		})
+	}
+}
+
+func generateEvent(t *testing.T) {
+	// Run a gadget so it's captured by the bpstats gadget
+	opts := gadgetrunner.GadgetRunnerOpts[ExpectedBpfstatsEvent]{
+		Image:   testGadgetImage,
```

Comment:
Mmmm even if we pass the name here, given that we are using the local runtime, it will be ignored, right?

TODO: Validate it once we support setting name for "local" instances. Related to https://github.com/inspektor-gadget/inspektor-gadget/pull/3820#discussion_r2035553258.